### ISSUE TITLE
Screen Density get lowered 

### ISSUE THREAD

**Issue by Nobody9090** [Mahmud0808/Iconify#341]
### Describe the bug

Ok, so I originally had the same issue as the other fellow. This time I went into the app and told it upon reboot, restart systemui after boot. Now. It didnt lower the Density nearly as bad. It was the ONLY thing I did before rebooting. Now, I turn the logger off by default in developer options. I am going to turn it back on.  And reboot and the issue should still persist. So this will be updated. But since its late and im tired, I am starting this. I am working on it now.

### Steps to reproduce

Basically after fixing the last issue density was lowered after doing what I did above

### Expected behavior

Density should not be lowered

### Screenshots

More to come with logs. Just starting this

### Device information

Pixel 7 PRO, offical Android 14 rooted with Magisk 

### Additional context

WIP

**Comment by Mahmud0808** [comment]
Are you by any chance using PixelXpert (formerly known as AOSPMods)?

**Comment by Nobody9090** [comment]
> Are you by any chance using PixelXpert (formerly known as AOSPMods)?

I am. I just got done looking at the logs. I am pretty sure I see the issue.

**Comment by Mahmud0808** [comment]
> > Are you by any chance using PixelXpert (formerly known as AOSPMods)?
> 
> I am.

In misc settings of PixelXpert, set display resolution to 100%

**Comment by Nobody9090** [comment]
> > > Are you by any chance using PixelXpert (formerly known as AOSPMods)?
> > 
> > 
> > I am.
> 
> In misc settings of PixelXpert, set display resolution to 100%

mActivity and mContext device profiles have different values, add more logs.
11-12 04:15:21.015  3198  3198 D b/254119092: 	mActivity logs:
11-12 04:15:21.015  3198  3198 D b/254119092: 		mActivity=null
11-12 04:15:21.015  3198  3198 D b/254119092: 	mContext logs:
11-12 04:15:21.015  3198  3198 D b/254119092: 		mContext=android.window.WindowContext@8d9a67d
11-12 04:15:21.015  3198  3198 D b/254119092: 		mContext.getResources().getConfiguration()={1.15 310mcc260mnc [en_US] ldltr sw823dp w823dp h1641dp 560dpi xlrg long hdr widecg port night finger -keyb/v/h -nav/h winConfig={ mBounds=Rect(0, 0 - 2880, 6240) mAppBounds=Rect(0, 288 - 2880, 6030) mMaxBounds=Rect(0, 0 - 2880, 6240) mDisplayRotation=ROTATION_0 mWindowingMode=fullscreen mDisplayWindowingMode=fullscreen mActivityType=undefined mAlwaysOnTop=undefined mRotation=ROTATION_0} as.56 s.9336 fontWeightAdjustment=0}
11-12 04:15:21.015  3198  3198 D b/254119092: 		LauncherAppState.getIDP().getDeviceProfile(mContext).isTaskbarPresent=true
11-12 04:15:21.015  3198  3198 D b/254119092: ComponentCallbacks#onConfigurationChanged() configDiffForRecreate={CONFIG_WINDOW_CONFIGURATION}
11-12 04:15:21.015  3198  3198 D b/254119092: mActivity and mContext device profiles have different values, add more logs.
11-12 04:15:21.015  3198  3198 D b/254119092: 	mActivity logs:
11-12 04:15:21.015  3198  3198 D b/254119092: 		mActivity=null
11-12 04:15:21.015  3198  3198 D b/254119092: 	mContext logs:
11-12 04:15:21.015  3198  3198 D b/254119092: 		mContext=android.window.WindowContext@8d9a67d
11-12 04:15:21.015  3198  3198 D b/254119092: 		mContext.getResources().getConfiguration()={1.15 310mcc260mnc [en_US] ldltr sw823dp w823dp h1641dp 560dpi xlrg long hdr widecg port night finger -keyb/v/h -nav/h winConfig={ mBounds=Rect(0, 0 - 2880, 6240) mAppBounds=Rect(0, 288 - 2880, 6030) mMaxBounds=Rect(0, 0 - 2880, 6240) mDisplayRotation=ROTATION_0 mWindowingMode=fullscreen mDisplayWindowingMode=fullscreen mActivityType=undefined

**Comment by Nobody9090** [comment]
I think that solves it. I dually apologize after scanning the logs. It is NOT your work or fault.

**Comment by Nobody9090** [comment]
11-12 04:15:21.033  2520  2995 V WindowManagerShell:  Attempt to merge sync (#41)android.os.BinderProxy@ea3546b@0 into (#40)android.os.BinderProxy@1e04959@1 via a SLEEP proxy
11-12 04:15:21.033  2520  2995 V WindowManagerShell:  Attempt to merge sync (#41)android.os.BinderProxy@ea3546b@0 into (#40)android.os.BinderProxy@1e04959@1 via a SLEEP proxy
11-12 04:15:21.034  2520  2995 E ShellTransitions: Trying to finish a non-running transition. Either remote crashed or  a handler didn't properly deal with a merge. (#39)android.os.BinderProxy@67bf7a7@1
11-12 04:15:21.034  2520  2995 E ShellTransitions: java.lang.RuntimeException
11-12 04:15:21.034  2520  2995 E ShellTransitions: 	at com.android.wm.shell.transition.Transitions.onFinish(go/retraceme 4bf4a24a8f53c97051f13f334d4dec3198658e2a941842270b262cc2ee0f4958:33)
11-12 04:15:21.034  2520  2995 E ShellTransitions: 	at com.android.wm.shell.transition.Transitions$$ExternalSyntheticLambda6.onTransitionFinished(go/retraceme 4bf4a24a8f53c97051f13f334d4dec3198658e2a941842270b262cc2ee0f4958:15)
11-12 04:15:21.034  2520  2995 E ShellTransitions: 	at com.android.wm.shell.transition.DefaultMixedHandler$$ExternalSyntheticLambda1.onTransitionFinished(go/retraceme 4bf4a24a8f53c97051f13f334d4dec3198658e2a941842270b262cc2ee0f4958:29)
11-12 04:15:21.034  2520  2995 E ShellTransitions: 	at com.android.wm.shell.keyguard.KeyguardTransitionHandler$1$$ExternalSyntheticLambda0.run(go/retraceme 4bf4a24a8f53c97051f13f334d4dec3198658e2a941842270b262cc2ee0f4958:17)
11-12 04:15:21.034  2520  2995 E ShellTransitions: 	at android.os.Handler.handleCallback(Handler.java:958)
11-12 04:15:21.034  2520  2995 E ShellTransitions: 	at android.os.Handler.dispatchMessage(Handler.java:99)
11-12 04:15:21.034  2520  2995 E ShellTransitions: 	at android.os.Looper.loopOnce(Looper.java:205)
11-12 04:15:21.034  2520  2995 E ShellTransitions: 	at android.os.Looper.loop(Looper.java:294)
11-12 04:15:21.034  2520  2995 E ShellTransitions: 	at android.os.HandlerThread.run(HandlerThread.java:67)
11-12 04:15:21.035  2520  2995 V WindowManagerShell: Transition animation finished (aborted=false), notifying core (#40)android.os.BinderProxy@1e04959@1
11-12 04:15:21.046  3198  3198 D b/254119092: DisplayController#onConfigurationChanged: {1.15 310mcc260mnc [en_US] ldltr sw823dp w823dp h1641dp 560dpi xlrg long hdr widecg port night finger -keyb/v/h -nav/h winConfig={ mBounds=Rect(0, 0 - 2880, 6240) mAppBounds=Rect(0, 288 - 2880, 6030) mMaxBounds=Rect(0, 0 - 2880, 6240) mDisplayRotation=ROTATION_0 mWindowingMode=fullscreen mDisplayWindowingMode=fullscreen mActivityType=undefined mAlwaysOnTop

More of it


If you have Density issues with Iconify, it is absolutely because you are having a train wreck with windowmanager because of using both apps

**Comment by Mahmud0808** [comment]
> I think that solves it. I dually apologize after scanning the logs. It is NOT your work or fault.

So the issue is solved by changing display resolution from PixelXpert?

**Comment by Nobody9090** [comment]
> > I think that solves it. I dually apologize after scanning the logs. It is NOT your work or fault.
> 
> So the issue is solved by changing display resolution from PixelXpert?

I posted more of the log. I dont think its because you all may use different values. The issue is because the binder you're getting back is an instance of BinderProxy, not your local binder class.

**Comment by Nobody9090** [comment]
https://github.com/Mahmud0808/Iconify/blob/cb7f5c6e8499b439264ab6903f134c83dd006057/app/src/main/java/com/drdisagree/iconify/services/BackgroundService.java#L27

and 

https://github.com/siavash79/PixelXpert/blob/82f346938703ab6597b3f240dd0a1af947b701f2/app/src/main/java/sh/siava/pixelxpert/modpacks/android/PackageManager.java#L10

**Comment by Nobody9090** [comment]
Its not Iconify that is screwing up the density on phones, it's a conflict with android.os.Binder and Ibinder returning a proxy

**Comment by Mahmud0808** [comment]
> https://github.com/Mahmud0808/Iconify/blob/cb7f5c6e8499b439264ab6903f134c83dd006057/app/src/main/java/com/drdisagree/iconify/services/BackgroundService.java#L27

This is not used anymore in iconify. I just kept the class but used it nowhere. It was last used maybe almost a year ago. That's why I added @Deprecated

**Comment by Nobody9090** [comment]
> > https://github.com/Mahmud0808/Iconify/blob/cb7f5c6e8499b439264ab6903f134c83dd006057/app/src/main/java/com/drdisagree/iconify/services/BackgroundService.java#L27
> 
> This is not used anymore in iconify. I just kept the class but used it nowhere. It was last used maybe almost a year ago. That's why I added @deprecated

Ah, I haven't done much with coding in some time. I had faith, you were right about your app. I'm 100% sure it is not your app. I will send you the complete log to an email address. I'm way too tired to go through it and make sure there is no identifying markers. I was already exhausted. If you want it. You can email me at steaven.android@gmail.com. I would like to take this moment and make sure this goes no further and that everything about your app is right. It is the mixing of the two that creates the issue and I take this time to apologize. There are some other errors from Iconify. Like I said if you want the log is yours.

**Comment by Mahmud0808** [comment]
> > > https://github.com/Mahmud0808/Iconify/blob/cb7f5c6e8499b439264ab6903f134c83dd006057/app/src/main/java/com/drdisagree/iconify/services/BackgroundService.java#L27
> > 
> > 
> > This is not used anymore in iconify. I just kept the class but used it nowhere. It was last used maybe almost a year ago. That's why I added @deprecated
> 
> Ah, I haven't done much with coding in some time. I had faith, you were right about your app. I'm 100% sure it is not your app. I will send you the complete log to an email address. I'm way too tired to go through it and make sure there is no identifying markers. I was already exhausted. If you want it. You can email me at [steaven.android@gmail.com](mailto:steaven.android@gmail.com). I would like to take this moment and make sure this goes no further and that everything about your app is right. It is the mixing of the two that creates the issue and I take this time to apologize. There are some other errors from Iconify. Like I said if you want the log is yours.

Sure you can find my email [HERE](https://github.com/Mahmud0808#connect-with-me).

**Comment by Nobody9090** [comment]
> > > > https://github.com/Mahmud0808/Iconify/blob/cb7f5c6e8499b439264ab6903f134c83dd006057/app/src/main/java/com/drdisagree/iconify/services/BackgroundService.java#L27
> > > 
> > > 
> > > This is not used anymore in iconify. I just kept the class but used it nowhere. It was last used maybe almost a year ago. That's why I added @deprecated
> > 
> > 
> > Ah, I haven't done much with coding in some time. I had faith, you were right about your app. I'm 100% sure it is not your app. I will send you the complete log to an email address. I'm way too tired to go through it and make sure there is no identifying markers. I was already exhausted. If you want it. You can email me at [steaven.android@gmail.com](mailto:steaven.android@gmail.com). I would like to take this moment and make sure this goes no further and that everything about your app is right. It is the mixing of the two that creates the issue and I take this time to apologize. There are some other errors from Iconify. Like I said if you want the log is yours.
> 
> Sure you can find my email [HERE](https://github.com/Mahmud0808#connect-with-me).

Again this is a dead issue. It shall never be spoken of again. Thank you for your patience. Great work by the way.
